### DISCUSSION TITLE
No package for Debian-based OS

### DISCUSSION THREAD

**Discussion by Pierstoval** [ImageMagick/ImageMagick#6720] [Contribute]
I've been using ImageMagick for a long time for my projects as a regular user (nothing very advanced).

However, I use it on Debian and Ubuntu and every time I want to use version `7.x` I have to recompile it from source.

It's slow as hell, and I wish there were `.deb` packages for that, or a native `imagemagick7` package.

On Debian there is currently a package for ImageMagick `6.9.10.23`, but nothing for `7.x`.

If I can build ImageMagick7 from source on each machine (especially my Docker images based on Debian), and there already is a native package for version 6, why isn't there a move on creating a package for version 7 for all recent Debian versions?

**Comment by urban-warrior** [comment]
We recommend you post your request to the Debian/Ubuntu developers.  We have no control over their choices for which releases of ImageMagick they distribute.

**Comment by Pierstoval** [comment]
@urban-warrior I understand this part, I wasn't sure about apt packages.

However, this does not answer the question about why there is no `.deb` package in the proposed downloads, as debian-based OS are very popular 😕

**Comment by urban-warrior** [comment]
Got it.  We're a small development team, and as such we have too much to do and too little time.  Instead, we rely on users such as yourself to contribute to the ImageMagick open source project with security flaws, patches, suggestions, and yes, even .deb configuration files :-).

**Comment by Pierstoval** [comment]
No worry, I'm not forcing anyone nor want to insist 😉 I was just pointing one fact, and I actually don't know how to create a `.deb` file at all, I just asked the question.

I may gather some information on how to create such package, but I don't guarantee anything 😄

**Comment by Jacob-Stevens-Haas** [comment]
Would love this too.  The version of Imagemagick that ships with ubuntu doesn't have `magick`, just `convert`, which doesn't seem to support file globbing.

Building from source now, but I don't really know enough about compilation to make it work if I need to link extra libraries, or something like that.

**Comment by Pierstoval** [comment]
@urban-warrior Is it possible to commission paid work from the IM team for this specific `.deb` feature? 😉

**Comment by urban-warrior** [comment]
Post the requirements of the `.deb` feature for us to review.

**Comment by Vinfall** [comment]
For reference, you can check [the Packaging page on Debian Wiki](https://wiki.debian.org/Packaging).

As for the packaging tools, ~~there used to be a handy tool namely `checkinstall` which makes packaging easy but it hasn't been updated since 2017 and is currently only available in Debian testing/unstable repo.~~ a quick research about the tools used in [Package/Learn](https://wiki.debian.org/Packaging/Learn) suggests that it should be enough if you install `devscripts` and `dpkg-dev` from the Debian repo. Most tools are bundled with these two packages. There is also a dedicated section listing the suggested tools in the packaging page.

For a quick assessment on the possible efforts you may put in, you may skim the slides-based [packaging tutorial](https://www.debian.org/doc/devel-manuals#packaging-tutorial).

**Comment by mviereck** [comment]
There is a nice script called [IMEI](https://github.com/SoftCreatR/imei) that makes it easy to compile ImageMagick.

Around IMEI I wrote a script [im7deb-docker](https://github.com/mviereck/im7deb-docker) to build ImageMagick within a docker container that finally results in a deb package.
It only depends on `docker` or `podman` to create a deb package. The desired debian or ubuntu release can be set optionally.

So far the resulting deb package works for me (debian bullseye here). However, it does not fulfill debian standards. It is just convenient to have a single installable and removeable package.

**Comment by habetsm-xilinx** [comment]
I see the latest Debian version is now 7.1.2.27.
https://packages.debian.org/search?keywords=ImageMagick
That is not far behind the latest version.
